spec/helpers/appstore_project_helper.rb

```diff
@@ -0,0 +1,214 @@
+            suffix.gsub!(forbidden_characters, '_')
+          end
+          suffix
+        end
+      end
+
+      def initialize(project, opts = { client: GoodData.connection })
+        @client = opts[:client]
+        @project = project if project.instance_of? GoodData::Project
+        @project = client.projects(project) if project.instance_of? String
+        GoodData.logger.info "Project ID: #{@project.obj_id}"
+        raise 'ProjectHelper::initialize the parameter "project" '\
+            'must be GoodData::Project or String.' unless @project
+      end
+
+      def create_ldm
+        json = File.read('./spec/data/integration_model.json')
+        blueprint = GoodData::Model::ProjectBlueprint.from_json(json)
+        project.update_from_blueprint(blueprint)
+      end
+
+      def load_data
+        GoodData.with_project @project.pid do |project|
+          GoodData::Model.upload_data(
+            './spec/data/integration_policies.csv',
+            project.blueprint,
+            DATASET_IDENTIFIER
+          )
+        end
+      end
+
+      def add_user_group
+        @project.add_user_group(:name => USER_GROUP_NAME, :description => 'My Test Description')
+      end
+
+      def associate_output_stage(ads, opts = {})
+        @project.create_output_stage(
+          ads,
+          project: @project,
+          client: client,
+          client_id: opts[:client_id],
+          output_stage_prefix: opts[:output_stage_prefix]
+        )
+      end
+
+      def deploy_processes(ads)
+        GoodData.with_project @project.pid do |project|
+          cc_process = project.deploy_process(
+            CC_PROCESS_ARCHIVE,
+            name: CC_PROCESS_NAME
+          )
+          cc_schedule = cc_process.create_schedule(
+            CC_SCHEDULE_CRON,
+            CC_GRAPH,
+            params: CC_PARAMS,
+            hidden_params: CC_SECURE_PARAMS,
+            state: 'DISABLED'
+          )
+          ruby_process = project.deploy_process(
+            RUBY_HELLO_WORLD_PROCESS_PATH,
+            name: RUBY_HELLO_WORLD_PROCESS_NAME
+          )
+          ruby_schedule = ruby_process.create_schedule(
+            cc_schedule,
+            'main.rb',
+            params: RUBY_PARAMS,
+            hidden_params: RUBY_SECURE_PARAMS,
+            state: 'DISABLED'
+          )
+
+          associate_output_stage ads
+          add_process = project.add.process
+          add_process.create_schedule(
+            ruby_schedule,
+            '',
+            dataload_datasets: [ DATASET_IDENTIFIER ],
+            de_synchronize_all: true,
+            state: 'DISABLED'
+          )
+        end
+      end
+
+      def create_metrics
+        GoodData.with_project @project.pid do |project|
+          metrics = JSON.parse(File.read(METRICS_FILE)).to_hash
+          metrics.each do |title, maql|
+            metric = project.add_measure maql, title: title
+            metric.identifier = "metric.#{title.to_s.downcase.gsub(/\s/, '.')}"
```

Comment:
Refers to 'metric' more than self (maybe move it to another class?) (FeatureEnvy)